lib/rubocop/cop/lint/interpolation_check.rb

```diff
@@ -35,6 +35,10 @@ def on_str(node)
 
         private
 
+        def regex_type?(parent)
+          parent && (parent.dstr_type? || parent.regexp_type?)
+        end
```

Comment:
This is to overcome the method complexity offense raised by rubocop

Reply:
Hmm, I think we can use safe navigation to simplify this a little bit

```ruby
def regex_type?(parent)
  parent&.dstr_type? || parent&.regexp_type?
end
```